src/GitVersionCore/OutputVariables/VariableProvider.cs

```diff
@@ -56,7 +57,7 @@ public VersionVariables GetVariablesFor(SemanticVersion semanticVersion, Effecti
             var semverFormatValues = new SemanticVersionFormatValues(semanticVersion, config);
 
             var informationalVersion = CheckAndFormatString(config.AssemblyInformationalFormat, semverFormatValues,
-                environment, semverFormatValues.DefaultInformationalVersion, "AssemblyInformationalVersion");
+                environment, semverFormatValues.DefaultInformationalVersion, "AssemblyInformationalVersion").RegexReplace("[^0-9A-Za-z-.+]", "-");
```

Comment:
Is it not possible for `CheckAndFormatString` to perform the `RegexReplace`? The problem can in theory occur for `assemblyFileSemVer` and `assemblySemVer` as well if their formats or schemes include the prerelease label (tag)?

Reply:
Hmm.
I see someone removed the fields assembly-versioning-format and assembly-file-versioning-format from the documentation but not from the code.

I would say it not needed to move the regex because for assembly version and assembly file version the rules are way more restricting.
Basicly only the schemes described in https://github.com/GitTools/GitVersion/blob/0a64063fd57099e6ffb96b8bbc5fc731a530aae5/src/GitVersionCore/Extensions/AssemblyVersioningScheme.cs#L5-L9
and 
https://github.com/GitTools/GitVersion/blob/0a64063fd57099e6ffb96b8bbc5fc731a530aae5/src/GitVersionCore/Extensions/AssemblyFileVersioningScheme.cs#L5-L9
are allowed by Microsoft specs.

I guess that is also the reason why the fields are no longer in the documentation for GitVersion 4.x and 5.x.

Also see [Microsoft docs](https://docs.microsoft.com/en-us/dotnet/standard/assembly/versioning)

Reply:
Good point. Would moving the `RegexReplace` hurt, though? I would prefer it inside the method if it doesn't actually destroy anything.

Reply:
Nop would not hurt i guess.
Just a theoretical performance hit. But i don't think that will be noticeable.